### ISSUE TITLE
Why don't the macros defined in .cljs files work in lumo?

### ISSUE THREAD

**Issue by philoskim** [anmonteiro/lumo#377]
I read http://blog.fikesfarm.com/posts/2015-06-19-portable-macro-musing.html about self-hosted ClojureScript macros.

The following is the tree structure for these experiments in lumo 1.8.0.

```
$ tree example/
example/
├── macro1.cljc
├── macro2.cljs
├── macro3.clj
├── test1.cljs
├── test2.cljs
└── test3.cljs
```

Firstly I experimented with it like this.

```
;; example/macro1.cljc
(ns example.macro1)

(defmacro my-calc [a b]
  #?(:clj  `(+ ~a ~b)
     :cljs `(* ~a ~b)))
```

```
;; example/test1.cljs
(ns example.test1
  (:require-macros [example.macro1 :refer [my-calc]]))

(println (my-calc 10 20))

```

```
$ lumo example/test1.cljs 
200
```

`:cljs` reader conditional part is run as I expected. 

However I cannot understand what follows.

```
;; example/macro2.cljs
(ns example.macro2)

(defmacro my-calc [a b]
   `(+ ~a ~b))
```

```
;; example/test2.cljs
(ns example.test2
  (:require-macros [example.macro2 :refer [my-calc]]))

(println (my-calc 10 20))

```

```
$ lumo example/test2.cljs 
No such macros namespace: example.macro2, could not locate example/macro2.clj or example/macro2.cljc
	 (new)
	 Function.cljs.core.ex_info.cljs$core$IFn$_invoke$arity$3 (NO_SOURCE_FILE <embedded>:2024:72)
	 Function.cljs.analyzer.error.cljs$core$IFn$_invoke$arity$3 (NO_SOURCE_FILE <embedded>:2639:92)
	 Function.cljs.analyzer.error.cljs$core$IFn$_invoke$arity$2 (NO_SOURCE_FILE <embedded>:2638:92)
	 (NO_SOURCE_FILE <embedded>:5952:162)
	 Object.lumo.repl.load_other (NO_SOURCE_FILE <embedded>:6471:378)
	 lumo.repl.load (NO_SOURCE_FILE <embedded>:6475:96)
	 Function.cljs.js.require.cljs$core$IFn$_invoke$arity$5 (NO_SOURCE_FILE <embedded>:5954:77)
	 Object.cljs.js.load_macros (NO_SOURCE_FILE <embedded>:5988:418)
	 (NO_SOURCE_FILE <embedded>:6011:316)

```

lumo doesn't run the macro defined in the .cljs file. However, it should work according to the article in http://blog.fikesfarm.com/posts/2015-06-19-portable-macro-musing.html (quote: With the bootstrapped ClojureScript compiler (let's call it ClojureScript JS), custom macros will perforce need to be written in ClojureScript, but still kept separate as is required by ClojureScript JVM.).  

Thirdly I experimented with it in the .clj file. it runs without any problem as indicated in the article.

```
;; example/macro3.clj
(ns example.macro3)

(defmacro my-calc [a b]
   `(+ ~a ~b))
```

```
;; example/test3.cljs
(ns example.test3
  (:require-macros [example.macro3 :refer [my-calc]]))

(println (my-calc 10 20))

```

```
$ lumo example/test3.cljs 
30
```
In conclusion, the macros defined in .clj or .cljc files are run but the macros in .cljs files aren't. I'm so confused about these experiments. What is the criteria to define macros in lumo? And what I want to know is whether this policy(the macros defined in .clj or .cljc files are run but the macros in .cljs files aren't.) will continue or change in the future. If I am misunderstanding, please let me know about it.

**Comment by RickMoynihan** [comment]
Interesting... also at a lumo REPL:

```
13:02 $ lumo
Lumo 1.8.0
ClojureScript 1.9.946
Node.js v9.2.0
 Docs: (doc function-name-here)
       (find-doc "part-of-name-here")
 Source: (source function-name-here)
 Exit: Control+D or :cljs/quit or exit

cljs.user=> (defmacro foo [form] (println "macro executing with form: " form))
#'cljs.user/foo
cljs.user=> (foo {:some :form})
macro executing with form:  nil
nil
```

**Comment by arichiardi** [comment]
There might be a problem in the way we load files, it can be classified as bug because I think it is supposed to work. However I have not investigated further.

**Comment by anmonteiro** [comment]
@RickMoynihan your example is working as expected. Read more about that here http://blog.fikesfarm.com/posts/2015-09-07-messing-with-macros-at-the-repl.html

**Comment by anmonteiro** [comment]
@philoskim We currently only look for macros in `.cljc` and `.clj` files (see https://github.com/anmonteiro/lumo/blob/6ca2a260c8a77ec7a7a0edeb41b36d13fa9b7d1f/src/cljs/snapshot/lumo/repl.cljs#L76).

The reason being that's how it works in JVM ClojureScript and we didn't want to fragment the ecosystem by having something that only works in self-hosted. AFAIK Planck also works the same way.

I'm not sure if this is a bug. Currently your example is supposed to work like you described.

**Comment by anmonteiro** [comment]
Question has been answered, closing.
